components/eamxx/src/physics/cosp/eamxx_cosp.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef SCREAM_COSP_HPP
+#define SCREAM_COSP_HPP
+
+#include "share/atm_process/atmosphere_process.hpp"
+#include "ekat/ekat_parameter_list.hpp"
+#include "ekat/ekat_pack_kokkos.hpp"
+#include "share/atm_process/ATMBufferManager.hpp"
+
+#include <string>
+
+namespace scream
+{
+
+/*
+ * The class responsible to handle the calculation of COSP diagnostics
+ * The AD should store exactly ONE instance of this class stored
+ * in its list of subcomponents (the AD should make sure of this).
+*/
+
+class Cosp : public AtmosphereProcess
+{
+
+public:
+
+  // Constructors
+  Cosp (const ekat::Comm& comm, const ekat::ParameterList& params);
+
+  // The type of subcomponent
+  AtmosphereProcessType type () const { return AtmosphereProcessType::Physics; }
+
+  // The name of the subcomponent
+  std::string name () const { return "cosp"; }
+
+  // Set the grid
+  void set_grids (const std::shared_ptr<const GridsManager> grids_manager);
+
+  // Scratch space for local variables
+  struct Buffer {
```

Comment:
Is this needed? If not, I would prune it.

Reply:
Removed.